webapi/src/main/twirl/org/knora/webapi/messages/twirl/queries/sparql/v1/findLinkValueByIri.scala.txt

```diff
@@ -45,44 +45,37 @@ SELECT ?linkValue ?directLinkExists ?targetResourceClass ?objPred ?objObj
     FROM <http://www.ontotext.com/explicit>
 }
 WHERE {
-    BIND(IRI("@subjectIri") AS ?subject)
-    BIND(IRI("@predicateIri") AS ?predicate)
-    BIND(IRI("@{predicateIri}Value") AS ?predicateForLinkValue)
-
-    @maybeObjectIri match {
-        case Some(objectIri) => {
-
-            BIND(IRI("@objectIri") AS ?object)
+    BIND(IRI("@linkValueIri") AS ?linkValue)
```

Comment:
In line 50 and 72 below, you are using `<@linkValueIri>` not the `?linkValue`, is the bind here really necessary? 
I see that `?linkValue` is used in `Select`, so if for that purpose BIND is necessary, why not use `?linkValue` in the other statements?

Reply:
The bind is there only so we can return `?linkValue` in `SELECT`. Using an IRI literal instead of a variable improves performance, so I changed all the variables to literals wherever possible in these queries.